universe/auto_syncer.go

```diff
@@ -531,6 +519,147 @@ func (f *FederationEnvoy) SetAllowPublicAccess() error {
 	)
 }
 
+// tryFetchServers attempts to fetch the set of universe servers in the
+// federation.
+func (f *FederationEnvoy) tryFetchServers() ([]ServerAddr, error) {
+	ctx, cancel := f.WithCtxQuit()
+
+	fedServers, err := f.cfg.FederationDB.UniverseServers(
+		ctx,
+	)
+	if err != nil {
+		cancel()
+
+		err := fmt.Errorf("unable to fetch set of universe servers: %v",
+			err)
+		f.reportErr(err)
+		return nil, fmt.Errorf("unable to fetch set of universe" +
+			"servers")
+	}
+	cancel()
+
+	return fedServers, nil
+}
+
+// FetchAssetInfo queries the universes in our federation for genesis and asset
+// group information about the given asset ID.
+func (f *FederationEnvoy) FetchAssetInfo(ctx context.Context,
+	assetID *asset.ID) error {
+
+	if assetID == nil {
+		return fmt.Errorf("no asset ID provided")
+	}
+
+	// Fetch the set of universe servers in our federation.
+	fedServers, err := f.tryFetchServers()
+	if err != nil {
+		return err
+	}
+
+	assetConfig := FedUniSyncConfig{
+		UniverseID: Identifier{
+			AssetID:   *assetID,
+			ProofType: ProofTypeIssuance,
+		},
+		AllowSyncInsert: true,
+		AllowSyncExport: false,
+	}
+	fullConfig := SyncConfigs{
+		UniSyncConfigs: []*FedUniSyncConfig{&assetConfig},
+	}
+	log.Tracef("universe config for asset lookup: %v",
+		spew.Sdump(assetConfig))
+
+	// We'll sync with Universe servers in parallel and collect the diffs
+	// from any successful syncs. There can only be one diff per server, as
+	// we're only syncing one universe root.
+	returnedSyncDiffs := make(chan AssetSyncDiff, len(fedServers))
+
+	// To fetch information about the asset, we only need to sync with the
+	// remote universe. Asset group import and verification is handled as
+	// part of the universe sync.
+	syncFromUni := func(ctxs context.Context, addr ServerAddr) error {
+		syncDiff, err := f.cfg.UniverseSyncer.SyncUniverse(
+			ctxs, addr, SyncIssuance, fullConfig,
+		)
+
+		// Sync failures are expected from Universe servers that do not
+		// have a relevant universe root.
+		if err != nil {
+			log.Debugf("asset lookup for %v failed with remote"+
+				"server: %v", assetID.String(), addr.HostStr())
+			return nil
+		}
+
+		// There should only be one sync diff since we're only syncing
+		// one universe root.
+		if syncDiff != nil {
+			if len(syncDiff) != 1 {
+				log.Debugf("unexpected number of sync diffs: "+
+					"%v", len(syncDiff))
+				return nil
+			}
+
+			returnedSyncDiffs <- syncDiff[0]
+		}
+
+		return nil
+	}
+
+	// Sync with the federation Universe servers in parallel.
```

Comment:
Why do we need to sync from all the severs in parallel, when we only want a single asset ID here? 

Instead I think we can just select one (at random?), then try to sync with just that instance.

Reply:
Intent was that you get the 'latest' asset state, e.x. for an asset group one server may only have record of 5 issuances while the official issuer server has a record of 10.

Also just to mirror what we do for the background sync. I think random single sync may not work well if we expect users to have universes with non-overlapping assets in their federation (one host wallet provider, one explorer, one stablecoin issuer, etc.).